atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleNamespaceDeleterIntegrationTest.java

```diff
@@ -0,0 +1,483 @@
+                .addAllTableReferences(listAllTablesWithMetadata())
+                .build();
+    }
+
+    private Set<String> listAllPhysicalTableNames(String tablePrefix, String overflowTablePrefix) {
+        String escapedTablePrefix = withWildcardSuffix(withEscapedUnderscores(tablePrefix));
+        String escapedOverflowTablePrefix = withWildcardSuffix(withEscapedUnderscores(overflowTablePrefix));
+
+        return runWithConnection(connection -> {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT table_name FROM all_tables WHERE owner = upper(?) AND (table_name LIKE upper(?)"
+                            + " ESCAPE '\\' OR table_name LIKE upper(?) ESCAPE '\\')");
+            statement.setString(1, dbKeyValueServiceConfig.connection().getDbLogin());
+            statement.setString(2, escapedTablePrefix);
+            statement.setString(3, escapedOverflowTablePrefix);
+
+            ResultSet resultSet = statement.executeQuery();
+            return getTableNamesFromResultSet(resultSet);
+        });
+    }
+
+    private Set<TableReference> listAllTablesWithMetadata() {
+        return runWithConnection(connection -> {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT table_name from " + AtlasDbConstants.DEFAULT_ORACLE_METADATA_TABLE);
+            ResultSet resultSet = statement.executeQuery();
+            return getTableNamesFromResultSet(resultSet).stream()
+                    .map(TableReference::createUnsafe)
+                    .collect(Collectors.toSet());
+        });
+    }
+
+    private Set<String> listAllTablesWithMapping() {
+        return runWithConnection(connection -> {
+            PreparedStatement statement =
+                    connection.prepareStatement("SELECT table_name from " + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE);
+            ResultSet resultSet = statement.executeQuery();
+            return getTableNamesFromResultSet(resultSet);
+        });
+    }
+
+    private <T> T runWithConnection(FunctionCheckedException<Connection, T, SQLException> task) {
+        try (Connection connection = connectionManager.getConnection()) {
+            return task.apply(connection);
+        } catch (SQLException e) {
+            throw Throwables.rewrapAndThrowUncheckedException(e);
+        }
+    }
+
+    private static Set<String> getTableNamesFromResultSet(ResultSet resultSet) throws SQLException {
+        ImmutableSet.Builder<String> builder = ImmutableSet.builder();
+        while (resultSet.next()) {
+            builder.add(resultSet.getString("table_name"));
+        }
+        return builder.build();
+    }
+
+    private static String withEscapedUnderscores(String s) {
+        return s.replace("_", "\\_");
+    }
+
+    private static String withWildcardSuffix(String s) {
+        return s + "%";
+    }
+
+    private static void assertSnapshotContainsNewTableReferences(
```

Comment:
maybe not actionable: I don't really like this on the basis of it checks more than what it says (it checks that the snapshot is a 'normal' looking snapshot with the given number of TRs). Though I don't really have a better name